src/shogun/mathematics/linalg/ratapprox/logdet/opfunc/LogRationalApproximationCGM.cpp

```diff
@@ -80,21 +67,30 @@ CJobResultAggregator* CLogRationalApproximationCGM::submit_jobs(
 		Map<VectorXcd> negated_shifts(m_negated_shifts.vector, m_negated_shifts.vlen);
 		negated_shifts=-shifts;
 	}
-
-	// create one CG-M job for current sample vector which solves for all
-	// the shifts, and computes the final result and stores that in the aggregator
-	CRationalApproximationCGMJob* job
-			=new CRationalApproximationCGMJob(agg, m_linear_solver,
-			m_linear_operator, sample, m_negated_shifts, m_weights, m_constant_multiplier);
-	SG_REF(job);
-
-	m_computation_engine->submit_job(job);
-
-	// we can safely unref the job here, computation engine takes it from here
-	SG_UNREF(job);
-
-	SG_DEBUG("Leaving\n");
-	return agg;
+	REQUIRE(
+		m_linear_operator->get_dimension() == sample.vlen,
+		"Dimension mismatch! %d vs %d\n", m_linear_operator->get_dimension(),
+		sample.vlen);
+	REQUIRE(
+		m_negated_shifts.vlen == m_weights.vlen,
+		"Number of shifts and weights are not equal!\n");
+
+	SGVector<complex128_t> vec = m_linear_solver->solve_shifted_weighted(
+		m_linear_operator, sample, m_negated_shifts, m_weights);
+
+	Map<VectorXcd> v(vec.vector, vec.vlen);
+	v = -v;
+	// take out the imaginary part of the result before
+	// applying linear operator
+	SGVector<float64_t> agg = m_linear_operator->apply(vec.get_imag());
+
+	// perform dot product
+	Map<VectorXd> map_agg(agg.vector, agg.vlen);
```

Comment:
yeah no... i mean i was already about to comment above that why do we use eigen here directly... but now seeing that it's being used to do the dot product below.... plz use `linalg::` over SGVector/Matrix

Reply:
I have used linalg in all opfunc classes however the ```DenseMatrixExactJob``` class use ```mat.log()``` which is not in linalg so it still uses Eigen3